api/server/sdk/volume_ops.go

```diff
@@ -968,6 +968,21 @@ func (s *VolumeServer) mergeVolumeSpecs(vol *api.VolumeSpec, req *api.VolumeSpec
 		spec.ProxySpec = vol.GetProxySpec()
 	}
 
+	// Pure NFS Endpoint
+	if vol.ProxySpec != nil && vol.ProxySpec.PureFileSpec != nil {
+		// if volume to be updated is Pure File Direct Access volume
+		if spec.ProxySpec == nil {
+			spec.ProxySpec = &api.ProxySpec{
+				PureFileSpec: &api.PureFileSpec{},
+			}
+		}
+		if req.GetPureNfsEndpoint() != "" {
```

Comment:
order has to be reversed, first check for req.GetPureNfsEndpoint() and then check vol. Here vol might be nil.

Reply:
Done